src/validator.spec.ts

```diff
@@ -0,0 +1,246 @@
+import { validator, setErrorMessage } from './validator';
+
+describe('isNumber validator', () => {
+  it('should return failed validation when it feeds value equals undefined', () => {
+    // Arrange
+    const value = void 0;
+
+    // Act
+    const result = validator({ value });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: 'Must be a number',
+      type: 'IS_NUMBER',
+    });
+  });
+
+  it('should return failed validation when it feeds value equals null', () => {
```

Comment:
I think for null, undefined or blank string validation should pass, if you want to control that add a required validator on the beginning of the list.

Why this behavior? I add a non mandatory field that has to be a number.

Reply:
You are right. I will change it

Reply:
Done